### DISCUSSION TITLE
Data Objects not visible in Object Listing and search results

### DISCUSSION THREAD

**Discussion by Piotrulaa** [pimcore/pimcore#12192] [Q&A / Support] [answered]
Hi, I encountered some problems when filling Pimcore with products data.

I created some data objects (like products, categories) and assets (images). My product class has panel with fields for image and category (many-to-many object relation). Problem happens when I try to add image/category value to their field in object edit view. Search results for objects/assets are empty, "No items found", despite everything is set as published.

Second problem I have is with DataHub Graphql API. I can retrieve single object by its id using query getObject, but when i try to retrieve all objects using query getObjectListing i receive an empty list.

Here is example of queries I made:
```
{
  getProduct(id: 2) {
    Title
  }
  
  getProductListing(first: 10) {
    totalCount
    edges {
      node {
        Title
      }
    }
  }
}
```

For me it looks like objects and assets "can't see each other", but I can't troubleshoot it.

Does anyone have idea what could cause such problems?

**Comment by kingjia90** [comment, marked as answer]
Have you already tried to run the `pimcore:search-backend-reindex` command?

**Reply by Piotrulaa** [reply]
Thanks a lot, it solved problem with empty search results. Unfortunately problem with empty list returned by getObjectListing DataHub query still occurs.

**Reply by kingjia90** [reply]
That is great to hear!

About the GraphQL issue, i will give a further look and let you know

**Reply by Piotrulaa** [reply]
Okay, thanks. Then I'm waiting :)

**Reply by kingjia90** [reply]
@Piotrulaa About the GraphQL, this could be related to your case https://github.com/pimcore/data-hub/issues/513
Is your datahub version 1.3.2+?

**Reply by Piotrulaa** [reply]
Where can I check version of my datahub?

**Reply by kingjia90** [reply]
🔧 Tools > Bundle & Bricks

**Reply by Piotrulaa** [reply]
Thanks, so: I have DataHub version 1.2.5

**Reply by kingjia90** [reply]
That may explain the problem, i would suggest to give a try to reproduce the problem with a more recent datahub, at least 1.3.2. If i am not wrong, it should fix and work properly.
Please open an new issue in DataHub repository with more details, if the problem still occurs even after the upgrade.

Hope it helps and wish you a nice weekend!

**Reply by Piotrulaa** [reply]
Ok, I will make update and check if it helps. Thanks a lot for all the help.
Have a nice weekend!

**Reply by Lam1na** [reply]
Sorry for being such a fool, but where do I have to run the code pimcore:search-backend-reindex? How can I do that? Console or file? Which file etc.? Can somebody tell me detailed how it works? Thank you all a lot :-)

**Reply by Piotrulaa** [reply]
[Here](https://pimcore.com/docs/pimcore/10.4/Development_Documentation/Development_Tools_and_Details/Console_CLI.html) are docs for this.

**Reply by Piotrulaa** [reply]
You will find this command when you run `./bin/console list`
